pydantic_ai_slim/pydantic_ai/models/groq.py

```diff
@@ -541,9 +543,14 @@ async def _get_event_iterator(self) -> AsyncIterator[ModelResponseStreamEvent]:
                 except IndexError:
                     continue
 
+                provider_details_dict: dict[str, Any] = {}
                 if raw_finish_reason := choice.finish_reason:
-                    self.provider_details = {'finish_reason': raw_finish_reason}
+                    provider_details_dict['finish_reason'] = raw_finish_reason
                     self.finish_reason = _FINISH_REASON_MAP.get(raw_finish_reason)
+                if self._provider_timestamp is not None:  # pragma: no branch
+                    provider_details_dict['timestamp'] = number_to_datetime(self._provider_timestamp)
```

Comment:
Since this is in the loop and we never reset `self._provider_timestamp`, we'll be calling `number_to_datetime` on every chunk!

We only need to store it if it's not there already. And we can likely do that outside of the loop as well.